### ISSUE TITLE
Finalize API

### ISSUE THREAD

**Issue by minecrawler** [Download/mics#2]
Continued discussion from #1 :

## @Download 's proposal was

```js
/**
 * Mix classes
 * Any class can be transformed to a mixin (by calling `.mix()`,
 * but can also be instantiated on its own
 */
class Mics {
  /**
   * Create a mixin from a class
   * Multiple classes or mixins can be mixed into the resulting mixin
   *
   * @param {MixinFactory} superclass, returns a class extending the first parameter
   * @param {...MixinFactory} mixins, return a class extending the first parameter
   * @param {function} factory
   * @return {object}
   */
  static mix(superclass, ...mixins, factory) {}

  /**
   * Check if an object is of a certain type or
   * was created with a certain mixin
   *
   * @param {object} obj
   * @param {class|string} mixin
   * @return {boolean}
   */
  static is(obj, mixin) {}

  /**
   * Check if an object can be used as if it was a certain type or
   * created with a certain mixin
   *
   * @param {object} obj
   * @param {class} mixin
   * @return {boolean}
   */
  static like(obj, mixin) {}
}

/**
 * This callback creates a class-extending class
 *
 * @callback MixinFactory
 * @param {class} superclass
 * @return {class}
 */
```

example usage:

```js
import { mix, is, like } from 'mics'

var viewer = {              // create an object with the 
  look(){}                  // same interface as Looker 
}     
is(viewer, Looker)          // false, but 
like(viewer, Looker)        // true 
```

---

## My proposal was

```js
/**
 * Mix classes
 * Any class can be transformed to a mixin (by calling `.mix()`,
 * but can also be instantiated on its own
 */
class Mics {
  /**
   * Create a mixin from a class
   * Multiple classes or mixins can be mixed into the resulting mixin
   *
   * @param {MixinFactory} superclass, returns a class extending the first parameter
   * @param {...MixinFactory} mixins, return a class extending the first parameter
   * @param {function} factory
   * @return {object}
   */
  static mix(superclass, ...mixins, factory) {}

  /**
   * Check if an object was created with a certain mixin
   *
   * @param {object} obj
   * @param {class} mixin
   * @return {boolean}
   */
  static hasMixin(obj, mixin) {}

  /**
   * Check if an object can be used as if it was created with a certain mixin
   *
   * @param {object} obj
   * @param {class} mixin
   * @return {boolean}
   */
  static likeMixin(obj, mixin) {}
}

/**
 * This callback creates a class-extending class
 *
 * @callback MixinFactory
 * @param {class} superclass
 * @return {class}
 */
```

Both are pretty similar, thought @Download 's exposes a bit more functionality in the same amount of methods.

**Comment by Download** [comment]
@minecrawler wrote:
> So, what's left is to consolidate the API and update the unit tests. As soon as you did that, I could set up the mixin stuff on my side and give you some feedback on the feeling and if I need different functionality and edge-cases I encounter. Well, I probably made that sound too easy.

Ha ha no let's not make it bigger than it is 👍 
I'll work on this and try to publish a 0.6 with the new API this weekend.

**Comment by Download** [comment]
Released in [0.6.0](https://github.com/Download/mics/releases/tag/0.6.0).

**Comment by Download** [comment]
One part is still missing: [Define which (if any) properties should be exported on mixins](https://github.com/Download/mics/issues/7)
